src/transformers/generation/utils.py

```diff
@@ -1245,6 +1245,46 @@ def _validate_model_kwargs(self, model_kwargs: Dict[str, Any]):
                 " generate arguments will also show up in this list)"
             )
 
+    def _validate_generation_length(self, generation_config, input_ids_length, has_default_max_length):
+        """Performs validation related to the resulting generated length"""
+
+        # 1. Max length warnings related to poor parameterization
+        if has_default_max_length and generation_config.max_new_tokens is None and generation_config.max_length != 20:
+            # 20 is the default max_length of the generation config
+            warnings.warn(
+                f"Using the model-agnostic default `max_length` (={generation_config.max_length}) to control the"
+                "generation length. We recommend setting `max_new_tokens` to control the maximum length of the "
+                "generation.",
+                UserWarning,
+            )
+        if input_ids_length >= generation_config.max_length:
+            input_ids_string = "decoder_input_ids" if self.config.is_encoder_decoder else "input_ids"
+            logger.warning(
+                f"Input length of {input_ids_string} is {input_ids_length}, but `max_length` is set to"
+                f" {generation_config.max_length}. This can lead to unexpected behavior. You should consider"
+                " increasing `max_new_tokens`."
+            )
+
+        # 2. Min length exceptions due to unfeasible parameter combinations
+        min_length_error_suffix = " To continue, decrease the minimum length and/or increase the maximum length."
+        if has_default_max_length:
+            min_length_error_suffix += (
+                f" Note that `max_length` is set to {generation_config.max_length}, its default value."
+            )
```

Comment:
This should make it clear when any minimum length argument clashes with the default value (whose existence is often unknown by the users)